thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/AlertFilterAutoTuneOnboardingTask.java

```diff
@@ -64,37 +69,41 @@ public void run() {
     Preconditions.checkNotNull(executionContext.getExecutionResult(ALERT_FILTER_FACTORY));
     Preconditions.checkNotNull(executionContext.getExecutionResult(ALERT_FILTER_AUTOTUNE_FACTORY));
 
-    AlertFilterFactory alertFilterFactory = (AlertFilterFactory) executionContext.getExecutionResult(ALERT_FILTER_FACTORY);
-    AlertFilterAutotuneFactory alertFilterAutotuneFactory = (AlertFilterAutotuneFactory)
-        executionContext.getExecutionResult(ALERT_FILTER_AUTOTUNE_FACTORY);
+    AlertFilterFactory alertFilterFactory =
+        (AlertFilterFactory) executionContext.getExecutionResult(ALERT_FILTER_FACTORY);
+    AlertFilterAutotuneFactory alertFilterAutotuneFactory =
+        (AlertFilterAutotuneFactory) executionContext.getExecutionResult(ALERT_FILTER_AUTOTUNE_FACTORY);
 
     Preconditions.checkNotNull(alertFilterFactory);
     Preconditions.checkNotNull(alertFilterAutotuneFactory);
 
-    DetectionJobResource detectionJobResource = new DetectionJobResource(new DetectionJobScheduler(),
-        alertFilterFactory, alertFilterAutotuneFactory);
+    DetectionJobResource detectionJobResource =
+        new DetectionJobResource(new DetectionJobScheduler(), alertFilterFactory, alertFilterAutotuneFactory);
 
-    AnomalyFunctionDTO anomalyFunctionSpec = (AnomalyFunctionDTO) executionContext.getExecutionResult(
-        ANOMALY_FUNCTION_CONFIG);
+    AnomalyFunctionDTO anomalyFunctionSpec =
+        (AnomalyFunctionDTO) executionContext.getExecutionResult(ANOMALY_FUNCTION_CONFIG);
     long functionId = anomalyFunctionSpec.getId();
     Period backfillPeriod = Period.parse(taskConfiguration.getString(BACKFILL_PERIOD, DEFAULT_BACKFILL_PERIOD));
     DateTime start = DateTime.parse(taskConfiguration.getString(BACKFILL_START, DateTime.now().toString()));
-    DateTime end = DateTime.parse(taskConfiguration.getString(BACKFILL_END, DateTime.now().minus(backfillPeriod).toString()));
+    DateTime end =
+        DateTime.parse(taskConfiguration.getString(BACKFILL_END, DateTime.now().minus(backfillPeriod).toString()));
 
     Response autotuneResponse = detectionJobResource.
         tuneAlertFilter(Long.toString(functionId), start.toString(), end.toString(),
             taskConfiguration.getString(AUTOTUNE_TYPE, DEFAULT_AUTOTUNE_TYPE),
             taskConfiguration.getString(HOLIDAY_STARTS, ""), taskConfiguration.getString(HOLIDAY_ENDS, ""),
-            taskConfiguration.getString(AUTOTUNE_FEATURES),
-            taskConfiguration.getString(AUTOTUNE_MTTD),
-            taskConfiguration.getString(AUTOTUNE_PATTERN, DEFAULT_AUTOTUNE_PATTERN));
+            taskConfiguration.getString(AUTOTUNE_FEATURES), taskConfiguration.getString(AUTOTUNE_MTTD),
+            taskConfiguration.getString(AUTOTUNE_PATTERN, DEFAULT_AUTOTUNE_PATTERN),
+            taskConfiguration.getString(AUTOTUNE_PATTERN_ONLY,
+                String.valueOf(Strings.isNullOrEmpty(taskConfiguration.getString(AUTOTUNE_MTTD)))));
```

Comment:
Use Boolean.toString() instead of String.valueOf()

Reply:
Thanks fixed.